packages/block-editor/src/components/block-list/block-list-compact.native.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * External dependencies
+ */
+import { View } from 'react-native';
+
+/**
+ * WordPress dependencies
+ */
+import { store as blockEditorStore } from '@wordpress/block-editor';
+import { useSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import styles from './style.scss';
+import BlockListBlock from './block';
+
+function BlockListCompact( props ) {
+	const {
+		marginHorizontal = styles.defaultBlock.marginLeft,
+		marginVertical = styles.defaultBlock.marginTop,
+		rootClientId,
+	} = props;
+	const { blockClientIds } = useSelect(
+		( select ) => {
+			const { getBlockOrder } = select( blockEditorStore );
+			const blockOrder = getBlockOrder( rootClientId );
+
+			return {
+				blockClientIds: blockOrder,
+			};
+		},
+		[ rootClientId ]
+	);
+
+	const containerStyle = {
+		marginVertical: -marginVertical,
+		marginHorizontal: -marginHorizontal,
+	};
+
+	return (
+		<View style={ containerStyle }>
+			{ blockClientIds.map( ( currentClientId ) => (
+				<BlockListBlock
+					clientId={ currentClientId }
+					key={ currentClientId }
+					marginHorizontal={ marginHorizontal }
+					marginVertical={ marginVertical }
```

Comment:
I noticed we're not including some of the props passed from the `InnerBlock` component that are used when rendering `BlockListBlock`:
https://github.com/WordPress/gutenberg/blob/1d9c05aa655d6039aca26980c89b1e23b6935598/packages/block-editor/src/components/inner-blocks/index.native.js#L121-L137
https://github.com/WordPress/gutenberg/blob/1d9c05aa655d6039aca26980c89b1e23b6935598/packages/block-editor/src/components/block-list/block-list-item.native.js#L139-L147

Especially the following ones:
- `parentWidth`: 
- `blockWidth`

Looks like these props are used for some style values calculations so not sure if we should also include them. As far as I tested the List V2, I haven't spotted any undesired side effects due to not including them but wanted to share it just in case.

Reply:
Ok, makes sense to keep it as is, especially if including those props implies adding more items to the React tree 😅.